horovod/runner/gloo_run.py

```diff
@@ -180,6 +186,9 @@ def _exec_command(command, slot_info, events):
                 stdout_file.close()
             if stderr_file:
                 stderr_file.close()
+        if exit_code != 0 and crash_event and not any(ev.is_set() for ev in all_events):
```

Comment:
Does it hurt to always set the `crash_event`, even if other events are set already`. We are not having such a logic for any of the other events.

Reply:
It doesn't hurt to set it, but it is very annoying to see this terminating message N times, since after one fails non-zero, they will all fail nonzero.

It would also be annoying if this message was printed after hitting ctrl+c, or something like that, because it would hide the real reason why things were terminating.  That is the purpose of checking all the other events.

From the user's perspective, it's very nice to know _why_ things are shutting down, and only emitting this message when we know it's true is very helpful.

Reply:
What about this logic then:
```
if exit_code != 0:
  if not crash_event.is_set():
    print('Terminating remaining workers after failure of Process {idx}.'.format(idx=index))
  crash_event.set()
```

Reply:
You would still get incorrect messages when any of the other events are already set, such as hitting `ctrl+c` (currently the only other way to have one of those events set).

In that case, it would be very confusing to tell the user that you are terminating workers as a result of a failed worker, because really the workers were already terminating due to some other event.